src/domain/transaction/evm/evm_transaction.rs

```diff
@@ -138,6 +138,15 @@ where
         &self.transaction_repository
     }
 
+    /// Checks if a provider error indicates the transaction was already submitted to the blockchain.
+    /// This handles cases where the transaction was submitted by another instance or in a previous retry.
+    fn is_already_submitted_error(error: &impl std::fmt::Display) -> bool {
+        let error_msg = error.to_string().to_lowercase();
+        error_msg.contains("already known")
```

Comment:
We should try to improve this. Not in this PR, but I prefer not to check if the error contains a message, even if we should do the same for the RPC Error msg.

Reply:
I agree.

Some parts are already checking error instance.
However some parts are trickier.